subprojects/base-services-groovy/src/main/java/org/gradle/api/specs/AndSpec.java

```diff
@@ -86,6 +86,12 @@ public boolean isSatisfiedBy(T object) {
         return uncheckedCast(EMPTY);
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
❌ Shouldn't override hashcode instead, but hash the class object in the superclass so that it is symmetric with its equals method.